alg/teca_temporal_index_select.cxx

```diff
@@ -0,0 +1,292 @@
+        return nullptr;
+    }
+
+    // get the input mesh
+    const_p_teca_mesh in_mesh
+        = std::dynamic_pointer_cast<const teca_mesh>(input_data[0]);
+
+    if (!in_mesh)
+    {
+        TECA_FATAL_ERROR("Failed to obtain indices; dataset is not a teca_mesh")
+        return nullptr;
+    }
+
+    // create the output mesh
+    p_teca_mesh out_mesh = std::static_pointer_cast<teca_mesh>
+        (std::const_pointer_cast<teca_mesh>(in_mesh)->new_shallow_copy());
+
+
+    // proceed if there are indices to remap; otherwise pass data along
+    if ( this->indices.size() > 0 )
+    {
+
+        // get the arrays and their names
+        p_teca_array_collection arrays = out_mesh->get_point_arrays();
+        std::vector<std::string> array_names;
+        request.get("arrays", array_names);
+        // pass the output mesh through if there are no arrays
+        if ( array_names.size() == 0 ) return out_mesh;
+
+
+        // loop over the point arrays
+        for ( auto& array_name : array_names)
+        {
+
+            // get the shape of one timestep of this array
+            const_p_teca_variant_array array_in = arrays->get(array_name);
+            size_t nxyz = array_in->size();
+
+            // set the shape of the output array
+            size_t nxyzt = nxyz * input_data.size();
+
+            // pre-allocate the output array
+            p_teca_variant_array array_out = array_in->new_instance();
+            // resize the output array so it can accommodate all timesteps
+            array_out->resize(nxyzt);
+            
+            // loop over the requests
+            for ( size_t i = 0; i < input_data.size(); ++i )
+            {
+                // get the current timestep's data
+                p_teca_mesh tmp_data = std::static_pointer_cast<teca_mesh>
+                    (std::const_pointer_cast<teca_mesh>(
+                    std::dynamic_pointer_cast<const teca_mesh>(input_data[i]))
+                    ->new_shallow_copy());
+
+                // get the current timestep's array
+                p_teca_variant_array tmp_array = 
+                    tmp_data->get_point_arrays()->get(array_name);
+
+                // insert data into the output array
+                array_out->set(i*nxyz, tmp_array, 0, nxyz);
+            }
+            // insert the new array into the output dataset
+            arrays->set(array_name, array_out);
+        }
+
+        // override the report of this slab
+        teca_metadata md_out = out_mesh->get_metadata();
+        // get the key for the time indices requested
+        std::string key;
+        if (request.get("index_request_key", key))
+        {
+            TECA_FATAL_ERROR("failed to get index request key")
+            return out_mesh;
+        }
+
+        // set the reported time extent to be the same as the request
+        std::vector<size_t> index_extent;
+        request.get(key, index_extent);
+        md_out.set(key, index_extent);
```

Comment:
There are also some convenience methods to do this.
https://teca.readthedocs.io/en/latest/doxygen/classteca__dataset.html#af9276ab5dc2067a00f767f811a7c8c7d

Reply:
to make sure I understand, you're suggesting replacing lines 284-287 with `out_mesh->set_request_indices(key, index_extent)`?

Reply:
Yes.

Reply:
cool, thanks for clarifying!